lisa/tools/curl.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from pathlib import PurePath
+from typing import Optional, cast
+
+from lisa.executable import Tool
+from lisa.operating_system import Posix
+from lisa.util.process import ExecutableResult
+
+
+class Curl(Tool):
+    @property
+    def command(self) -> str:
+        return "curl"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _install(self) -> bool:
+        posix_os: Posix = cast(Posix, self.node.os)
+        try:
+            posix_os.install_packages("curl")
+        except Exception as e:
+            self._log.debug(f"failed to install curl: {e}")
+        return self._check_exists()
+
+    def fetch(
+        self,
+        url: str,
+        arg: str,
+        execute_arg: str,
+        sudo: bool = False,
+        shell: bool = False,
+        cwd: Optional[PurePath] = None,
+    ) -> ExecutableResult:
+        err_msg = "curl fetch failed"
+        cmd_arg = f" {arg} {url}"
+        if execute_arg:
+            cmd_arg = f"{cmd_arg} | sh {execute_arg}"
+        result = self.run(
+            cmd_arg,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=err_msg,
+            sudo=sudo,
+            cwd=cwd,
+            shell=shell,
+        )
+        return result
+
+    def get_version(
+        self,
+        sudo: bool = False,
+        shell: bool = False,
+        cwd: Optional[PurePath] = None,
+    ) -> ExecutableResult:
+        err_msg = "curl get_version failed"
+        result = self.run(
```

Comment:
please check get_version defined in gcc, and update it accordingly.

Reply:
sure Lili